QRealTime_dialog.py

```diff
@@ -624,6 +620,82 @@ def getTable(self,XFormKey,importData,topElement,version= 'null'):
         except Exception as e:
             print ('not able to fetch',e)
             return response,table
+
+class Central (Aggregate):
+    tag="ODK Central"
+    # user auth token
+    usertoken = ""
+
+    def __init__(self,parent,caller):
+        super(Central, self).__init__(parent,caller)
+        
+    def setParameters(self):
+        self.parameters =[
+        ["id","Central"],
+        ["url",'https://sandbox.getodk.cloud'],
+        [self.tr("user"), ''],
+        [self.tr("password"), ''],
+        [self.tr("last Submission"),''],
+        [self.tr('sync time'),''],
+        [self.tr('project name'),'']
+        ]
+        
+    def getFormList(self):
+        user=self.getValue(self.tr("user"))
+        password=self.getValue(self.tr("password"))
+        c_url=self.getValue('url')
+        data = {'email': user, 'password' : password}
+        # todo: add check to see if url exists
+        headers = {"Content-Type": "application/json"}
+        projects = {}
+        forms = {}
+        project_name =self.getValue(self.tr("project name"))
+        project_id = 0
+        try:
+            x  = requests.post(c_url + "v1/sessions", json = data, headers = headers)
+            token = x.json()["token"]
+            global usertoken
+            usertoken = token
+            projects_response = requests.get(c_url + "v1/projects/", headers={"Authorization": "Bearer " + token})
+            # todo: add check to see if project name was entered
+            for p in projects_response.json():
+                if p["name"] == project_name:
+                    project_id = p["id"]
+            # todo: add message to say project name doesn't exist
+            form_response = requests.get(c_url + "v1/projects/"+ str(project_id)+"/forms/", headers={"Authorization": "Bearer " + token})
+            for form in form_response.json():
+                forms[form["name"]] = form["enketoOnceId"]
+            return forms, x
+        except:
+            self.iface.messageBar().pushCritical(self.tag,self.tr("Invalid url username or password"))
+            return None,None
+
+    def importData(self,layer,selectedForm,importData=True):
+        #from central 
+        user=self.getValue(self.tr("user"))
+        password=self.getValue(self.tr("password"))
+        c_url=self.getValue('url')
+        # todo: add check to see if url exists
+        para={'format':'xml'}
+        data = {'email': user, 'password' : password}
+        headers = {"Content-Type": "application/json"}
+        requests.packages.urllib3.disable_warnings()
+        try:
+            response = requests.get(c_url+'v1/projects/4/forms/geopoint-form.xml', headers ={"Authorization": "Bearer " + usertoken})

```

Comment:
why url is appended with v1/projects/4........... seems to be a hard coding.
It should read from ui